lib/trie.js

```diff
@@ -0,0 +1,77 @@
+
+module.exports = Trie;
+
+function Trie() {
+  this.methods = {};
+}
+
+// insert new nodes to Trie
+Trie.prototype.add = function(route, handler) {
```

Comment:
I'd personally use `verb` + `fullPath` arguments instead of `route`, e.g.

```js
Trie.prototype.add = function(verb, fullPath, handler) {
  // ...
};
```

That way the `Trie` is not coupled with strong-remoting's routes and also it's more obvious what data is needed for `add`.